src/app/pages/pass-finalisation/pass-finalisation.analytics.effects.ts

```diff
@@ -299,4 +308,58 @@ export class PassFinalisationAnalyticsEffects {
     }),
   ));
 
+  passFinalisationSeekFurtherDevelopment$ = createEffect(() => this.actions$.pipe(
+    ofType(SeekFurtherDevelopmentChanged),
+    concatMap((action) => of(action).pipe(
+      withLatestFrom(
+        this.store$.pipe(
+          select(getTests),
+        ),
+        this.store$.pipe(
+          select(getTests),
+          select(getCurrentTest),
+          select(getReview),
+          select(getFurtherDevelopment),
+        ),
+      ),
+    )),
+    concatMap((
+      [, tests, furtherDevelopment]: [ReturnType<typeof SeekFurtherDevelopmentChanged>, TestsModel, boolean],
+    ) => {
+      this.analytics.logEvent(
+        formatAnalyticsText(AnalyticsEventCategories.POST_TEST, tests),
+        formatAnalyticsText(AnalyticsEvents.FURTHER_DEVELOPMENT_CHANGED, tests),
+        `further development changed to ${furtherDevelopment ? 'Yes' : 'No'}`,
+      );
+      return of(AnalyticRecorded());
+    }),
+  ));
+
+  passFinalisationReasonGiven$ = createEffect(() => this.actions$.pipe(
+    ofType(ReasonForNoAdviceGivenChanged),
+    concatMap((action) => of(action).pipe(
+      withLatestFrom(
+        this.store$.pipe(
+          select(getTests),
+        ),
+        this.store$.pipe(
+          select(getTests),
+          select(getCurrentTest),
+          select(getReview),
+          select(getReasonForNoAdviceGiven),
+        ),
+      ),
+    )),
+    concatMap((
+      [, tests, reason]: [ReturnType<typeof ReasonForNoAdviceGivenChanged>, TestsModel, string],
+    ) => {
+      this.analytics.logEvent(
+        formatAnalyticsText(AnalyticsEventCategories.POST_TEST, tests),
+        formatAnalyticsText(AnalyticsEvents.REASON_FOR_NO_ADVICE_CHANGED, tests),
+        `reason for no advice changed ${reason}`,
```

Comment:
As this text repeats would it not be better to move it to an analytics label and make use of the 4th input for logEvent (value).  lower the amount of free text directly in the code

Reply:
Shall i take out the free text altogether? The second param probs explains it enough anyways